### ISSUE TITLE
Using commands in commands

### ISSUE THREAD

**Issue by Luis-Henriquez-Perez** [okamsn/loopy#112]
Sometimes I've wanted to use loopy commands in commands such as `initially`, `after-do`, and `finally`. This adds to my point in #109 that certain loopy commands are very useful even outside of a `loopy` loop.

I naturally want to do things like the below example.

```elisp
(loopy (list word '("I" "am" "sam"))
       (concating word :separator "\s")
       (after-do (concating ".")))

;; Would like it to return this.
;; => "I am sam."

;; But actually (ignoring the not yet existing `:separator' keyword) gives me this.
;; => void-function error concating.
```

**Comment by okamsn** [comment]
See #62, of which this is almost a duplicate. The challenge is that accumulation variables are finalized before code in `after-do` is run, such as correcting a list built in reverse. Any accumulation commands would have to be handled specially there, but at that point they're not much more convenient than just the functions that make them.

Ideally, `(concating ".")` is just equivalent to `(cl-callf concat loopy-result ".")`, which isn't (in my opinion) significantly more to type for a one-time thing. With how the code currently works, with implicit variables being optimized and explicit variables not, it would have to be used as `(concating loopy-result ".")` anyway.

The main benefit of the commands (again, in my opinion) is the automatic setup and finalizing of values that they do, neither of which are really needed in places like `after-do`.

**Comment by Luis-Henriquez-Perez** [comment]
> I think this is a duplicate of #62.

Sorry. That explains the dejavu feeling I had.

> Ideally, (concating ".") is just equivalent to (cl-callf concat loopy-result "."), which isn't (in my opinion) significantly more to type for a one-time thing.

The thing is many loops have that behavior where you do something repeatedly in the loop, then something slightly different at the end or beginning (which is why `before-do` and `after-do` and similar clauses are a good idea). So it might be a one-time thing for a particular loop that you're writing, but it will certainly happen again with some other loop.

To me the increased typing is not the only--or even the main--problem. It's the mental shift needed to go from `(concating ".")` to `(cl-callf concat loopy-result ".")`. After integrating the loopy syntax, and getting an error when trying to use it in `after-do` I think: "Why can't I do this? I just did it with `(concating word :separator "\s")`". It's a bit jarring for me.

This is partly why I make the argument that loopy accumulation commands are useful even outside of loops. For the "one-time" cases.

To be clear, my hopes in mentioning this is to provide user experience. I think one of loopy's goals is to make writing a loop fluid, intuitive and easy. So when I encounter something that seems suprising or jarring to me I mention it.

**Comment by okamsn** [comment]
> To me the increased typing is not the only--or even the main--problem. It's the mental shift needed to go from `(concating ".")` to `(cl-callf concat loopy-result ".")`. After integrating the loopy syntax, and getting an error when trying to use it in `after-do` I think: "Why can't I do this? I just did it with `(concating word :separator "\s")`". It's a bit jarring for me.

I hadn't considered this much. I will try to work on this after #107 and #109, which could be a while. 

I expect that one will still need to specify the target variable, since the assumption that optimized accumulation commands are written without the target variable is built into the commands' definitions (and likely other places in the code, unintentionally). Trying to avoid that would probably lead to a lot of duplication.
